pyanaconda/core/hw.py

```diff
@@ -0,0 +1,70 @@
+#
+# hw.py - utility functions dealing with hardware
+#
+# Copyright (C) 2022  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import blivet.arch
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+if blivet.arch.get_arch() in ["ppc64", "ppc64le"]:
+    MIN_RAM = 768
+    GUI_INSTALL_EXTRA_RAM = 512
+else:
+    MIN_RAM = 320
+    GUI_INSTALL_EXTRA_RAM = 90
+
+MIN_GUI_RAM = MIN_RAM + GUI_INSTALL_EXTRA_RAM
+SQUASHFS_EXTRA_RAM = 750
+NO_SWAP_EXTRA_RAM = 200
+
+
+def total_memory():
```

Comment:
I think we might be able to drop this and use `blivet.util.total_memory()` instead.

Reply:
I'll check that.

Reply:
There are no tests, no checks (but arguably one exception or the other matters little), and making the call requires importing things like six and bytesize. And it returns Size instead of int. All of that might be ok, though.

I wonder if this is an opportunity to contribute to blivet the extra bits?

Reply:
@vojtechtrefny would you be interested in the more checks and tests from https://github.com/rhinstaller/anaconda/pull/4338/commits/f22386efafe33df7f8182e65aaf99279904344c6 ? Your fork of the function looks like [this](https://github.com/storaged-project/blivet/blob/dff7867538c654d4d3fdcec8876e645050cae93f/blivet/util.py#L307).

Reply:
I think it is a good idea to open a PR for Blivet.

Reply:
> Replaced. Looks good.
> 
> @vojtechtrefny would you be interested in the [more checks and tests](https://github.com/rhinstaller/anaconda/blob/f22386efafe33df7f8182e65aaf99279904344c6/pyanaconda/core/hw.py#L37) for `total_memory` from [f22386e](https://github.com/rhinstaller/anaconda/commit/f22386efafe33df7f8182e65aaf99279904344c6) ? Your fork of the function looks like [this](https://github.com/storaged-project/blivet/blob/dff7867538c654d4d3fdcec8876e645050cae93f/blivet/util.py#L307).

Yes, that's a great idea.

Reply:
Done in https://github.com/storaged-project/blivet/pull/1073